### ISSUE TITLE
Cache the output of open source models

### ISSUE THREAD

**Issue by rlouf** [dottxt-ai/outlines#371]
We currently cache the results of requests to the OpenAI API. We should do the same thing with OSS models. The cache would only be effective if the same rng is passed at each call.

**Comment by avirlrma** [comment]
Hey @rlouf,
I was looking around in the codebase for this, and had question in terms of what do we want to cache for OSS model? Do we want to cache the sampled output or just the logits that are outputted by the model? The logits would be quite large so ideally we shouldn't be doing that. 

Also, just to confirm the if cache behaviour should be similar to openAI API call where we cache an entire generation at once instead of caching tokens (which I thought would be the way given current impl)?

**Comment by avirlrma** [comment]
Hi @rlouf ,
I am almost done with this. Will open a draft PR in a couple of days.
